src/random_read_worker.rs

```diff
@@ -95,11 +110,11 @@ impl RandomReadWorker {
                 let (qid, q) = self.query_gen.next_query();
 
                 let qfut = self.client.get(self.endpoint.clone()).query(&[("query", &q)]).send();
-                let dur = Self::measure_query(qfut)
+                let dur = Self::measure_query(self.behav, qfut)
                     .await
                     .map_err(|e| WorkerError::ReadFailed { query: q.into_owned(), err: e })?;
 
-                if let Some(id) = qid {
+                if let (Some(id), Some(dur)) = (qid, dur) {
```

Comment:
Is this needed to deal with `Ok(None)` from `measure_query`?

Reply:
Yeah, if either the query didn't get an id (because the `QueryGenerator` doesn't assign ids) or `measure_query` decided to ignore the error we have nothing to report.